### DISCUSSION TITLE
Cover issue with Home Assistant 2021.6

### DISCUSSION THREAD

**Discussion by garywatts** [Koenkk/zigbee2mqtt#7771] [General]
Hi, 
A breaking change in Home Assistant has caused problems with z2m covers. My Aqara B1 curtain motor are no longer available, It seems Home Assistant has changed the rules when using 'value_template' for covers and it should be changed to 'position_topic'. Where would one find where that bit of code is in z2m?

**Warning message prior to Home Assistant 2021.6**
"using 'value_template' for 'position_topic' is deprecated and will be removed from Home Assistant in version 2021.6, please replace it with 'position_template'"

**Error with 2021.6**
"2021-06-12 18:16:39 ERROR (MainThread) [homeassistant.util.logging] Exception in async_discover when dispatching 'mqtt_discovery_new_cover_mqtt': ({'availability': [{'topic': 'zigbee2mqtt/bridge/state'}], 'command_topic': 'zigbee2mqtt/Bed Front Window/set', 'device': {'identifiers': ['zigbee2mqtt_0x04cf8cdf3c8ee5c9'], 'manufacturer': 'Xiaomi', 'model': 'Aqara B1 curtain motor (ZNCLDJ12LM)', 'name': 'Bed Front Window', 'sw_version': 'Zigbee2MQTT 1.17.1-dev'}, 'json_attributes_topic': 'zigbee2mqtt/Bed Front Window', 'name': 'Bed Front Window', 'position_topic': 'zigbee2mqtt/Bed Front Window', 'set_position_template': '{ "position": {{ position }} }', 'set_position_topic': 'zigbee2mqtt/Bed Front Window/set', 'unique_id': '0x04cf8cdf3c8ee5c9_cover_zigbee2mqtt', 'value_template': '{{ value_json.position }}', 'platform': 'mqtt'},)
Traceback (most recent call last):
File "/usr/local/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 272, in __call__
return self._compiled([], data)
File "/usr/local/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 817, in validate_callable
return schema(data)
File "/usr/src/homeassistant/homeassistant/components/mqtt/cover.py", line 118, in validate_options
raise vol.Invalid(
voluptuous.error.Invalid: 'value_template' must be set together with 'state_topic'.
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/components/mqtt/mixins.py", line 160, in async_discover
config = schema(discovery_payload)
File "/usr/local/lib/python3.8/site-packages/voluptuous/validators.py", line 218, in __call__
return self._exec((Schema(val) for val in self.validators), v)
File "/usr/local/lib/python3.8/site-packages/voluptuous/validators.py", line 341, in _exec
raise e if self.msg is None else AllInvalid(self.msg, path=path)
File "/usr/local/lib/python3.8/site-packages/voluptuous/validators.py", line 337, in _exec
v = func(v)
File "/usr/local/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 276, in __call__
raise er.MultipleInvalid([e])
voluptuous.error.MultipleInvalid: 'value_template' must be set together with 'state_topic'."

**Comment by greynoO** [comment]
Same here. After a recent home assistant update my covers (ZNCLDJ11LM) don't work anymore in home assistant (entities for position and open/close do not exist with mqtt discovery). It works through the Admin panel of the zigbee2mqtt addon.

**Comment by garywatts** [comment]
FYI. It looks like this issue was fixed in a recent release 1.19 https://github.com/Koenkk/zigbee2mqtt/issues/7680

I did not get the update as the home assistant add on store repo has changed to https://github.com/zigbee2mqtt/hassio-zigbee2mqtt

**Reply by greynoO** [reply]
Thank you!
